src/macros.cr

```diff
@@ -67,6 +77,30 @@ macro record(name, *properties)
 
     {{yield}}
 
+    def copy_with({{
```

Comment:
I really think the method should be called `merge` since it mimic the `NamedTuple#merge` method.